src/Dfe.PlanTech.Application/Response/Commands/ProcessCheckAnswerDtoCommand.cs

```diff
@@ -0,0 +1,136 @@
+using Dfe.PlanTech.Application.Questionnaire.Queries;
+using Dfe.PlanTech.Application.Response.Interface;
+using Dfe.PlanTech.Application.Submission.Interface;
+using Dfe.PlanTech.Application.Submission.Interfaces;
+using Dfe.PlanTech.Domain.Questionnaire.Models;
+
+namespace Dfe.PlanTech.Application.Response.Commands
+{
+    public class ProcessCheckAnswerDtoCommand
+    {
+        private readonly IGetQuestionQuery _getQuestionQuery;
+        private readonly IGetAnswerQuery _getAnswerQuery;
+        private readonly GetQuestionQuery _getQuestionnaireQuery;
+        private readonly IGetResponseQuery _getResponseQuery;
+        private readonly ICalculateMaturityCommand _calculateMaturityCommand;
+
+        public ProcessCheckAnswerDtoCommand(
+            IGetQuestionQuery getQuestionQuery,
+            IGetAnswerQuery getAnswerQuery,
+            GetQuestionQuery getQuestionnaireQuery,
+            IGetResponseQuery getResponseQuery,
+            ICalculateMaturityCommand calculateMaturityCommand)
+        {
+            _getQuestionQuery = getQuestionQuery;
+            _getAnswerQuery = getAnswerQuery;
+            _getQuestionnaireQuery = getQuestionnaireQuery;
+            _getResponseQuery = getResponseQuery;
+            _calculateMaturityCommand = calculateMaturityCommand;
+        }
+
+        private async Task<CheckAnswerDto> _GetCheckAnswerDto(int submissionId)
+        {
+            var responseList = await _GetResponseList(submissionId);
+
+            CheckAnswerDto checkAnswerDto = new CheckAnswerDto();
+
+            Dictionary<string, int> indexMap = new Dictionary<string, int>();
+            Dictionary<string, DateTime> dateTimeMap = new Dictionary<string, DateTime>();
+
+            int index = 0;
+
+            foreach (Dfe.PlanTech.Domain.Responses.Models.Response response in responseList ?? throw new NullReferenceException(nameof(responseList)))
```

Comment:
This index variable is for keeping track of the QuestionsWithAnswers list specifically, so I can update an index if a later response with the same question appears without looping through the list, rather than tracking the current response array index itself

Reply:
E.g., if QuestionsWithAnswers[1] = QuestionOne, and responseList[1] and then responseList[8] = QuestionOne (changed answer at some point), I want to update QuestionsWithAnswers[1] with the information from responseList[8], hence the separate index variable